### DISCUSSION TITLE
Battery Powered

### DISCUSSION THREAD

**Discussion by Clem-** [jomjol/AI-on-the-edge-device#2556] [General]
Hi,
Thank you very much for your awesome project!!

I tried to run your project on a battery (18650 Li-ion cell) powered ESP and I've found out that it last several hours only.

I was wondering if there were some optimization to improve the power consumption like trying to disable the wifi interface and get in sleep between the camera shots? Does it make any sense?

**Comment by caco3** [comment]
Hi @Clem- 
The hardware is not made to be low power optimized. And there is nothing we can do against this.

Please search the discussions for similar discussions.

**Reply by monocycler** [reply]
Has anyone succeeded to add solar charging?  I am trying with MCP73871 USB 5V DC Solar Lipoly Lithium Lon Polymer Charger Board

**Reply by caco3** [reply]
you will need some decent buffering! Easiest might be to power it from a USB-battery pack which gets recharged from your solar panels.

**Reply by reichemn** [reply]
The hardware is not optimized but what the hardware offers is still not used by the software. I experimented with a battery powered AI-on-the-edge-device-device about a year ago. When the device idles between intervals, it still uses about 100mA@5V (via the USB interface). This can be drastically reduced to only a few milliamps if the ESP is sent to deep sleep for the remaining time of the interval. Unfortunately, I no longer have the exact measured values.
In combination with a low measuring frequency of only a few measurements per hour, this could enable a week (or even a month with very large power banks) of battery powered operation.

Since the device is not connected via WiFi during deep sleep, we have a difficult time to do configuration changes, especially with long sleep periods. The easiest way to solve this, would be to use an external trigger to enter deep sleep. I wish the device had a mqtt topic which could be used to tell the device "go to deep sleep for xxx seconds". This could be easily used with home assistance. Just create an automation which is triggered by the "flow finished" message in the status topic, which then sends a "go to sleep for 30 minutes" message to the device. Voila, you have a device sitting in deep sleep for most of the time. And when you need to change some configuration on the device, you could just disable this automation in home assistant, wait for the device to wake up, make your configuration changes, reenable the automation and send the device the sleep again.

**Reply by caco3** [reply]
There are some discussions about this.
@jomjol did some measurements and found that a large part of the power consumption comes from the camera. So even if the ESP goes to deep sleep, the cam continues to draw power. And the hardware has no way to switch the cam off.

In any way, if you want to implement a sleep mode for the ESP, feel free to create a pull request with such an implementation.

Regarding wakeup from sleep mode, there might be a solution:
Add a topic which gets set by another client (eg. homeassistant) and let it set the [retain flag](https://www.hivemq.com/blog/mqtt-essentials-part-8-retained-messages/). The ESP would have to periodically wake up and reconnect to the MQTT broker. After connection, it would get the last value of that subscribed topic and could eg. go back to sleep right away or stay awake.

**Reply by aleks-mariusz** [reply]
Just want to chime in on this and say I'm also highly interested in this! And I'm sure a lot of people are really in minimizing the power-usage.  Running on batteries is a _real-world_ need (esp when your meter is outdoors and running power to it is _not feasible_, whereas connecting a solar-charged battery-pack is muuuuuch more straight-forward).  

I would definitely appreciate more work being put into lowering the power usage to the limits available given the hardware limitations (of the camera constantly being powered and still using about 10mA during deep-sleep, that's still much less and requires no hardware modification.  One can explore that if they need something further, but the deep-sleep solution you guys mentioned is pretty brilliant!

**Reply by caco3** [reply]
@aleks-mariusz You might be interested to add your points to https://github.com/jomjol/AI-on-the-edge-device/discussions/2963
